source/slang/parser.cpp

```diff
@@ -347,11 +366,37 @@ namespace Slang
         }
     }
 
-    Token Parser::ReadToken()
+    TokenType Parser::PeekTokenType()
+    {
+        if (genericCloserCount)
```

Comment:
Some comments might be welcome here.

My understanding from looking at this is that the conceptual lookahead amounts to `genericCloserCount` instances of `>` tokens, followed by whatever `tokenReader.PeekTokenType()` would return.